test/libsolidity/natspecJSON/custom.sol

```diff
@@ -0,0 +1,43 @@
+/// @custom:x one two three
+/// @custom:y line
+/// break
+/// @custom:t one
+/// @custom:t two
+contract A {
+    /// @custom:note statevar
+    uint x;
+    /// @custom:since 2014
+    function g(int x) public pure virtual returns (int, int) { return (1, 2); }
+}
+
+// ----
+// ----
```

Comment:
Why are there two of these?

Reply:
That's the feature I added in #14505. `CommonSyntaxTest` now supports extra expectations after the error list and these are separated with an extra `// ----`. The base class stops reading from the stream there and the derived class can continue, interpreting the content any way it wants.

`NatspecJSONTest` interprets it as Natspec JSON. It does it inside `parseCustomExpectations()`, which is virtual and provided specifically so that the derived class can put such a logic there. We it could do this already in constructor, [like `CommonSyntaxTest` does](https://github.com/ethereum/solidity/blob/develop/test/CommonSyntaxTest.cpp#L62), though I don't really like that mechanism - it's too implicit and I feel like it would be too easy to mess up the order.

I could have used any other separator, but reusing this one seemed logical. The only downside is that we have two when the error section is empty. On the other hand, I don't have to worry about inventing a new one and making sure it's not used for something else.